orttraining/orttraining/python/experimental/amp/loss_scaler.py

```diff
@@ -27,12 +29,12 @@ def loss_scale(self):
 
     @loss_scale.setter
     def loss_scale(self, loss_scale):
-        assert isinstance(loss_scale, float) and loss_scale > 0, "'loss_scale' must be a positive float"
-        self._loss_scale = loss_scale
+        assert isinstance(loss_scale, (int, float)) and loss_scale > 0, "'loss_scale' must be a positive float"
```

Comment:
nit: should we remove float in the message since now that int is also allowed? `'loss_scale' must be positive`

Reply:
The right type is float and we want to enforce the best practices. The fact that we convert int to float was just an implementation detail that adds robustness